### ISSUE TITLE
Changing variable values in personal cache note does not update "variables" tab

### ISSUE THREAD

**Issue by dcjkfgdjhd** [cgeo/cgeo#13838]
### Describe your problem!

When you write things like "A=5" in the personal cache note, and then go to the "variables" tab, then c:geo automatically created a new variable called "A" and gives it the value "5".

However, if you then edit the personal cache note to, say, "A=7" then the variable in the "variables" tab keeps the old value "5".

This can lead to false calculations of the next stage, final, etc. and these errors are very hard to spot (you have to compare every variable in the PCN with the variables in the "variables" tab one by one). 

### How to reproduce?

See above.

### Reproducible

Yes

### c:geo Version

2022.12.21

**Comment by eddiemuc** [comment]
This is a wanted feature to prevent accidental deletion of manual edits in the variable tab. Synchronisation from personal note to variable tab is just a comfort feature. Likewise, synchronization of waypoints from note to waypoint tab will never delete preciously created waypoints or modify their data (if present).

**Comment by dcjkfgdjhd** [comment]
> This is a wanted feature to prevent accidental deletion of manual edits in the variable tab. Synchronisation from personal note to variable tab is just a comfort feature.

Well, but as it is currently implemented it leads to confusion, possible false calculations, and cleaning up that confusion is a lot of work for the user. Probably the vast majority of users simply write down their variable values into the PNC.

Maybe there should be a warning message along the lines of "There is currently a conflict between variables assignments. In the variable tab the variable A=5 is set, in the PCN the variable A=7. Which value do you want to keep?" and then the user can choose whether to keep A=5 or to change it to A=7?

> Likewise, synchronization of waypoints from note to waypoint tab will never delete preciously created waypoints or modify their data (if present).

I personally consider that a nuisance as well (I would prefer the extracted waypoint to get updated), but in this case there is an easy way out: one can simply create a new waypoint and keep the first one unchanged. Waypoints are unlimited, but variables unfortunately need unique values.

**Comment by eddiemuc** [comment]
I like the idea with a warning message displayes on sync when different values are found. The hard part is probably not to display it too often (e.g. if user has accepted once different values in var tab and notes, how to prevent that this is displayed over and over again on every note or variable value change?)

Also, one has to consider what should happen if the user defines the same var multiple times in the note e.g. like this:

```
This is my personal cache note
A=6
On second stage:
B=4
It seems that A=2 is more correct
```

**Comment by dcjkfgdjhd** [comment]
> if user has accepted once different values in var tab and notes, how to prevent that this is displayed over and over again on every note or variable value change?

I suppose c:geo would have to keep an internal record of the *last* values of all variables it extracted from the PCN, and if the PCN gets changed, but after this change the *extracted value of some variable remains the same*, then it should not ask again for this variable.

> Also, one has to consider what should happen if the user defines the same var multiple times in the note e.g. like this:

Maybe, c:geo could display a *list* of *all* possible assignments, that it has detected, and then let the user select one? That would 

Maybe using a list of radio buttons?

**Comment by eddiemuc** [comment]
> I suppose c:geo would have to keep an internal record of the last values of all variables it extracted from the PCN...

If this should be kept even across closing cache Details and reopening the cache this must even be persisted in database. This will make the change quite complex

**Comment by dcjkfgdjhd** [comment]
Sounds great to me! 👍

**Comment by eddiemuc** [comment]
Fixed by PR #13867
